crates/common_utils/src/types.rs

```diff
@@ -226,6 +232,70 @@ where
     }
 }
 
+/// Amount convertor trait for connector
+pub trait AmountConvertor: Send {
+    /// Output type for the connector
+    type Output;
+    /// helps in conversion of connector required amount type
+    fn convert(
+        &self,
+        i: MinorUnit,
+        currency: enums::Currency,
+    ) -> Result<Self::Output, TryFromIntError>;
+
+    /// helps in converting back connector required amount type to core minor unit
+    fn convert_back(
+        &self,
+        i: Self::Output,
+        currency: enums::Currency,
+    ) -> Result<MinorUnit, error_stack::Report<ParsingError>>;
+}
+
+/// Connector required amount type
+#[derive(Default, Debug, serde::Deserialize, serde::Serialize, Clone, Copy, PartialEq)]
+pub struct StringMajorUnitForConnector;
+
+impl AmountConvertor for StringMajorUnitForConnector {
```

Comment:
At connector level currently we use the terminology of `Base`, `Minor`. Either we would need to change it in connector Level, or change it here. But it should be consistent across the codebase.

Reply:
we are going to do these changes anyway in the connector module, will unify once its done for all the connector